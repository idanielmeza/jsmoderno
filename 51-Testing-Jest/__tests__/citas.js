import Citas from '../js/classes/Citas';

describe('Probar la clase de citas', ()=>{
    const id = Date.now();
    const citas = new Citas()
    
    test('Agregar nueva cita', ()=>{
        const citaObj = {
            id,
            mascota: 'Matias',
            propietario: 'Daniel',
            telefono: '321213',
            fecha: '10/12/2021',
            hora:'10:30',
            sintomas: 'Se come su caca'
        }

        citas.agregarCita(citaObj);

        expect(citas).toMatchSnapshot();

    })

    test('Actualizar cita', ()=>{
        const citaObjAct = {
            id,
            mascota: 'Matias Meza',
            propietario: 'Daniel',
            telefono: '321213',
            fecha: '10/12/2021',
            hora:'10:30',
            sintomas: 'Se come su caca'
        }

        citas.editarCita(citaObjAct);

        expect(citas).toMatchSnapshot();
    })

    test('Eliminar cita', ()=>{
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    })

})