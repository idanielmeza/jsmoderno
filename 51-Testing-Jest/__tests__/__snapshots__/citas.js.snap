// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10/12/2021",
      "hora": "10:30",
      "id": 1622501014926,
      "mascota": "Matias Meza",
      "propietario": "Daniel",
      "sintomas": "Se come su caca",
      "telefono": "321213",
    },
  ],
}
`;

exports[`Probar la clase de citas Agregar nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10/12/2021",
      "hora": "10:30",
      "id": 1622501014926,
      "mascota": "Matias",
      "propietario": "Daniel",
      "sintomas": "Se come su caca",
      "telefono": "321213",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar cita 1`] = `
Citas {
  "citas": Array [],
}
`;
